### PR TITLE
avm2: move event dispatch error handling in a single place

### PR DESCRIPTION
**Pull request by moulins**
The `Avm2::{dispatch, broadcast}_event` methods now log and swallow any AVM2 error occuring during the dispatch, instead of repeating the handling code for each caller.

This also introduces some behavioral changes:
- Errors messages are more consistent;
- For consistency with `broadcast_event`, `dispatch_event` panics if given a non-event object to dispatch.

### CODE REVIEW

**Review by Aaron1011 — APPROVED** [review]

**Review by Herschel — COMMENTED** [review]
Thank you!

Looks like `avm2::events::dispatch_event_to_target` still swallows any errors instead of propagating them up to `broadcast_event`/`dispatch_event`, and this is the method most enterFrame handlers etc. go through.

TODO someday: These errors should get fed to `loaderInfo.uncaughtErrorEvents`.

File: core/src/avm2.rs [hunk 1]
```diff
@@ -388,12 +402,18 @@ impl<'gc> Avm2<'gc> {
...
                 if object.is_of_type(on_type, &mut activation) {
-                    Avm2::dispatch_event(&mut activation.context, event, object)?;
+                    if let Err(err) = events::dispatch_event(&mut activation, object, event) {
+                        tracing::error!(
+                            "Encountered AVM2 error when broadcasting `{}` event: {}",
+                            event_name,
+                            err.detailed_message(&mut activation),
+                        );
+                        // Stop broadcasting after the first encountered error.
+                        return;
```
[the remark this answers is not in the document]
Comment (moulins) [reply core/src/avm2.rs]: I removed the incorrect early return, and added a 'TODO' comment for the `loaderInfo` stuff.

File: core/src/player.rs [hunk 2]
```diff
@@ -1426,40 +1418,35 @@ impl Player {
...
-                                {
-                                    tracing::error!(
-                                        "Encountered AVM2 error when broadcasting `progress` event: {}",
-                                        e
-                                    );
-                                }
-                            }
+                    match progress_evt {
+                        Err(e) => tracing::error!(
+                            "Encountered AVM2 error when dispatching `progress` event: {}",
```
Comment (Lord-McSweeney) [inline core/src/player.rs:1445]: I think this message should be changed to clarify that the error was encountered while trying to catch the event, for example ```"Encountered AVM2 error while constructing `progress` event: {}"```

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**